addon/window/browser.js

```diff
@@ -0,0 +1,35 @@
+import Base from './base';
```

Comment:
I pulled this code down and had some trouble mapping the methods between inherited relationships. Would this work just as well as 2 separate classes implementing the same interface? I have a hunch that this might be easier for me to understand. 

For instance I think a separate class would make it clear that the default value for `stubMediaFeatures` on the `NullWindow` class could be `{}`. Perhaps the `NullWindow` doesn't need these `if (this.stubbedMediaFeatures)` guards.